src/d3a/models/strategy/external_strategies/load.py

```diff
@@ -395,35 +395,54 @@ def event_tick(self):
         for req in self.pending_requests:
             if req.request_type == "set_energy_forecast":
                 self._set_energy_forecast_impl(req.arguments, req.response_channel)
+            elif req.request_type == "set_energy_measurement":
+                self._set_energy_measurement_impl(req.arguments, req.response_channel)
 
         self.pending_requests = deque(
             req for req in self.pending_requests
```

Comment:
Your proposal does not work, because we like to call the corresponding callback for each `request_type`. Also, I would not like to touch this code much because Spyros seems to have had problems with the MQTT subscriber.